### DISCUSSION TITLE
Unsuperivised learning for graph classification?

### DISCUSSION THREAD

**Discussion by karthickai** [pyg-team/pytorch_geometric#7049] [Q&A] [answered]
Hi all, Greetings. Could someone point me to a code example for using Unsupervised GraphSAGE or different architectures to generate node embeddings for downstream graph classification tasks?

**Comment by wsad1** [comment]
Check out this [example](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/graph_sage_unsup.py).

**Reply by karthickai** [reply]
Yes, I have seen this example already. However, this example doesn't suit for downstream graph classification tasks because it uses LinkNeighborLoader for the Cora dataset. Is there any direct example code to use the MUTAG kind of dataset for unsupervised learning using graphSAGE for graph classification? Thanks.

**Reply by rusty1s** [reply, marked as answer]
You can still use the obtained node embeddings for a downstream task by aggregating them together and applying an MLP on top.
